ui/components/app/flask/snaps-authorship-pill/snaps-authorship-pill.js

```diff
@@ -28,57 +45,92 @@ const SnapsAuthorshipPill = ({ snapId, version, className }) => {
   const url = isNPM
     ? `https://www.npmjs.com/package/${packageName}`
     : packageName;
-  const icon = isNPM ? 'fab fa-npm fa-lg' : 'fas fa-code';
-  const t = useI18nContext();
 
-  const friendlyName = getSnapName(snapId);
+  const subjectMetadata = useSelector((state) =>
+    getTargetSubjectMetadata(state, snapId),
+  );
+
+  const subjectName = subjectMetadata?.name ?? packageName;
+
+  const snapName = getSnapName(snapId);
+
+  const friendlyName = snapName === packageName ? subjectName : snapName;
+
+  const iconUrl = subjectMetadata?.iconUrl;
+
+  const fallbackIcon = friendlyName && friendlyName[0] ? friendlyName[0] : '?';
 
   return (
-    <a
-      href={url}
-      target="_blank"
-      rel="noopener noreferrer"
-      className={classnames(className, `snaps-authorship-pill`)}
+    <Box
+      className={classnames('snaps-authorship-pill', className)}
+      backgroundColor={BackgroundColor.backgroundDefault}
+      borderColor={BorderColor.borderDefault}
+      borderWidth={1}
+      alignItems={AlignItems.center}
+      paddingLeft={2}
+      paddingTop={2}
+      paddingBottom={2}
+      paddingRight={4}
+      borderRadius={BorderRadius.pill}
+      display={DISPLAY.FLEX}
+      style={{ maxWidth: 'fit-content', width: '100%' }}
     >
-      <Chip
-        leftIcon={
-          <Box paddingLeft={2}>
-            <i className={`${icon} snaps-authorship-icon`} />
-          </Box>
-        }
-        rightIcon={
-          version && (
-            <Box
-              className="snaps-authorship-version"
-              backgroundColor={BackgroundColor.primaryDefault}
-              paddingLeft={2}
-              paddingRight={2}
+      <Box>
+        <BadgeWrapper
+          badge={
+            <AvatarIcon
+              iconName={ICON_NAMES.SNAPS}
+              size={ICON_SIZES.XS}
+              backgroundColor={IconColor.infoDefault}
+              iconProps={{
+                size: ICON_SIZES.XS,
+                color: IconColor.infoInverse,
+              }}
+            />
+          }
+          position={BadgeWrapperPosition.bottomRight}
+        >
+          {iconUrl ? (
+            <AvatarFavicon size={Size.LG} src={iconUrl} />
+          ) : (
+            <AvatarBase
```

Comment:
Should we componentize this as e.g. `<SnapAvatar />` on a snaps-app-level so it can be used consistently across our UI? I have a feeling this will be used quite often going forward.

I know for example @david0xd will be needing it on the settings page as well.

Reply:
Yes, doing this now 👍